onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -543,14 +573,26 @@ static Status GetConvOpQuantizationScaleAndZeroPoint(
   if (weight_tensor.data_type() == ONNX_NAMESPACE::TensorProto_DataType_UINT8)
     return Status::OK();
 
-  // Now we have u8s8 QlinearConv
+  // This is per-tensor u8s8
+  // NNAPI does not support per-tensor u8s8
+  // For this case we will need to convert the int8 weight tensor to uint8
+  // And have same scale and 128 as zero point
+  // The coversion of the weight tensor itself will be done in the OpBuilder
+  const auto& scale_tensor = *initializers.at(input_defs[4]->Name());
+  int64_t scale_dim = scale_tensor.dims().empty() ? 1 : scale_tensor.dims()[0];
+  if (scale_dim == 1) {
+    w_zero_point = 128;
```

Comment:
If GetBinaryOpQuantizationScaleAndZeroPoint returned a w_zero_point that is not zero, what happens here?

Reply:
We verify before this can run in line 278 helper.cc

Reply:
I guess this is already checked by HasValidQuantizationZeroPoints?